services/dataService.js

```diff
@@ -0,0 +1,46 @@
+
+
+const products = require('../src/data/products');
+const categories = require('../src/data/categories');
+
+const productMap = {};
```

Comment:
initialize this as null, and check it for null to fetch the data

Reply:
@Goutam192002 did you see this? 
```suggestion
let productMap = null;
```